synapse/api/auth.py

```diff
@@ -173,6 +173,21 @@ async def get_user_by_req(
                     parent_span.set_tag("device_id", requester.device_id)
                 if requester.app_service is not None:
                     parent_span.set_tag("appservice_id", requester.app_service.id)
+
+                # Tag any headers that we need to extract from the request. This
+                # is useful to specify any headers that a reverse-proxy in front
+                # of Synapse may be sending to correlate and match up something
+                # in that layer to a Synapse trace. ex. when Cloudflare times
+                # out it gives a `cf-ray` header which we can also tag here to
+                # find the trace.
+                for header_name in self.hs.config.tracing.request_headers_to_tag:
+                    headers = request.requestHeaders.getRawHeaders(header_name)
```

Comment:
We use this pattern everywhere but is there a way to make this case insensitive?

For example, the Cloudflare docs mention `CF-Ray` but it seems to be `cf-ray` IRL. This could probably change at any time.

Reply:
Twisted handles this for you:

https://github.com/twisted/twisted/blob/7c00738a0c24032070ce92304b4c8b887666c0fc/src/twisted/web/http_headers.py#L114-L125

Reply:
hmmm, the only reason I mentioned this is because it didn't seem to work in my testing.

But trying it again now, it does. Must of been my typo 🤔